service/deferrer/service.go

```diff
@@ -31,3 +41,81 @@ func New(config Config) (*Service, error) {
 
 	return s, nil
 }
+
+// ShouldDefer finds corresponding DrainerConfig for the POD it's running in
+// and checks if node in current POD is drained yet. If DrainerConfig doesn't
+// exist or doesn't have Drained or Timeout condition, node termination should
+// be deferred.
+//
+// Current POD name and namespace are picked from environment variables with
+// corresponding keys defined in constants EnvKeyMyPodName &
+// EnvKeyMyPodNamespace. Defining these env variables is most conveniently
+// achieved by utilizing Kubernetes Downward API:
+// https://kubernetes.io/docs/tasks/inject-data-application/environment-variable-expose-pod-information/
+func (s *Service) ShouldDefer(ctx context.Context) (bool, error) {
+	s.logger.LogCtx(ctx, "level", "debug", "message", "finding out if node termination must be deferred")
```

Comment:
Super minor, the wording of the method is "should" and the sentiment of the logs is "must".